core/transaction-pool/src/lib.rs

```diff
@@ -20,10 +20,246 @@
 #![warn(unused_extern_crates)]
 
 mod api;
+mod maintainer;
 #[cfg(test)]
 mod tests;
 
 pub mod error;
 
-pub use api::{FullChainApi, LightChainApi};
 pub use txpool;
+pub use crate::api::{FullChainApi, LightChainApi};
+pub use crate::maintainer::{
+	TransactionPoolMaintainer,
+	DefaultFullTransactionPoolMaintainer, DefaultLightTransactionPoolMaintainer,
+};
+
+use std::{collections::HashMap, hash::Hash, sync::Arc};
+use futures::Future;
+
+use client::{
+	Client,
+	light::fetcher::Fetcher,
+	runtime_api::TaggedTransactionQueue,
+};
+use primitives::{Blake2Hasher, H256};
+use sr_primitives::{
+	serde::Serialize,
+	generic::BlockId,
+	traits::{Block as BlockT, Member, ProvideRuntimeApi},
+};
+
+use txpool::{BestIterator, EventStream, Options, watcher::Watcher};
+
+/// Extrinsic hash type for a pool.
+pub type ExHash<P> = <P as TransactionPool>::Hash;
+/// Block hash type for a pool.
+pub type BlockHash<P> = <<P as TransactionPool>::Block as BlockT>::Hash;
+/// Extrinsic type for a pool.
+pub type ExtrinsicFor<P> = <<P as TransactionPool>::Block as BlockT>::Extrinsic;
+
+/// Transaction pool interface.
+pub trait TransactionPool: Send + Sync {
```

Comment:
The trait should live outside of this crate, we don't want other implementations to depend on the entire pool.

Maybe put it to `primitives` or some `transaction-pool/api` or wherever other service-generic fields reside.

Reply:
Have made a separate crate - `transaction-pool/api` (will add a top-level comment about that)